vividus-plugin-avro/src/main/java/org/vividus/avro/steps/AvroSteps.java

```diff
@@ -0,0 +1,85 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.avro.steps;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.file.SeekableByteArrayInput;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.io.DatumReader;
+import org.jbehave.core.annotations.When;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.bdd.context.IBddVariableContext;
+import org.vividus.bdd.variable.VariableScope;
+import org.vividus.util.ResourceUtils;
+
+public class AvroSteps
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(AvroSteps.class);
+    private final IBddVariableContext bddVariableContext;
+
+    public AvroSteps(IBddVariableContext bddVariableContext)
+    {
+        this.bddVariableContext = bddVariableContext;
+    }
+
+    /**
+     * Converts the provided resource or file in Avro format to JSON and saves it as a text to a variable
+     *
+     * @param resourceNameOrFilePath The resource name or the file path.
+     * @param scopes                 The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of the variable
+     *                               scopes.<br>
+     *                               <i>Available scopes:</i>
+     *                               <ul>
+     *                               <li><b>STEP</b> - the variable will be available only within the step,
+     *                               <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                               <li><b>STORY</b> - the variable will be available within the whole story,
+     *                               <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                               </ul>
+     * @param variableName           The variable name to save resulting JSON.
+     * @throws IOException if an I/O error occurs
+     */
+    @When("I convert Avro data from `$resourceNameOrFilePath` to JSON and save result to $scopes variable "
+            + "`$variableName`")
+    public void convertAvroDataToJson(String resourceNameOrFilePath, Set<VariableScope> scopes, String variableName)
+            throws IOException
+    {
+        byte[] data = ResourceUtils.loadResourceOrFileAsByteArray(resourceNameOrFilePath);
+        DatumReader<Object> datumReader = new GenericDatumReader<>();
+        try (DataFileReader<Object> dataFileReader = new DataFileReader<>(new SeekableByteArrayInput(data),
+                datumReader))
+        {
+            LOGGER.atInfo()
+                    .addArgument(() -> dataFileReader.getSchema().toString(false))
+                    .log("Avro schema: {}");
+
+            StringBuilder json = new StringBuilder("[");
+
+            Object dataRecord = null;
```

Comment:
why is this variable needed?

Reply:
see line 78, it's used as a pointer in file